### PR TITLE
Internal source consumption related changes

### PR DESCRIPTION
**Pull request by kirankumarkolli**
Internal source consumption related changes
1. Ability to override through Directory.Build.Props
2. Conditional nuget package inclusion

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (kirankumarkolli) [inline Directory.Build.props]: How to enable conditional import?

File: Microsoft.Azure.Cosmos/src/Batch/PartitionKeyRangeBatchResponse.cs [hunk 2]
```diff
@@ -134,7 +134,12 @@ public override IEnumerator<BatchOperationResult> GetEnumerator()
             }
         }
 
-        internal override IEnumerable<string> GetActivityIds()
+#if INTERNAL
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Batch/PartitionKeyRangeBatchResponse.cs]: Why is it set public if INTERNAL is true?

File: Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj [hunk 3]
```diff
@@ -67,16 +65,18 @@
       <Generator>ResXFileCodeGenerator</Generator>
       <LastGenOutput>ClientResources.Designer.cs</LastGenOutput>
     </EmbeddedResource>
-  </ItemGroup>  
+  </ItemGroup>
+  
+  <ItemGroup Condition=" '$(ProjectRef)' != 'True' ">
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj]: In what scenarios will it be a project reference?

File: Microsoft.Azure.Cosmos/src/Batch/PartitionKeyRangeBatchResponse.cs (hunk 2 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Batch/PartitionKeyRangeBatchResponse.cs]: Its used by interop.

File: Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj (hunk 3 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj]: When ProjectRef is set to True, then packages are not included. And soruce needs to add project reference explicitly.

**Pull request merged**